### DISCUSSION TITLE
Example for building a subscription string

### DISCUSSION THREAD

**Discussion by richardryangarcia** [enisdenjo/graphql-ws#479] [General]
Hi all, 

I'm new to graphql-ws and just trying to use it for some basic graphql subscriptions. All the examples I have seen show a hard coded subscription string. Is there a recommended way to build that string from a Document Node? Maybe using graphql-request or graphql-tag? I really just want to be able to dynamically create the string by passing in a query (with params) and the variables for the params. 

Let me know if this makes sense or if I need to explain it any better. I appreciate the help!

**Comment by n1ru4l** [comment]
If you have a `DocumentNode` (AST), you can print it to a string using the `print` function from `graphql-js`. https://graphql.org/graphql-js/language/#print

```ts

const str = print(documentNode)

```
